### PR TITLE
Some minor cleanup

### PR DESCRIPTION
**Pull request by Ademan**
Happy to discuss any of the changes, I think they're pretty straight forward and uncontroversial, tests pass.

CI is broken for me, but main is also broken for me so I'm not certain that's relevant, or if I need to configure something in my fork repo.

### CODE REVIEW

**Comment by Ademan** [comment]
Actually I think the CI failure might be caused by the release of crossbeam-utils 0.8.17 yesterday.

**Comment by ulrichard** [comment]
> Actually I think the CI failure might be caused by the release of crossbeam-utils 0.8.17 yesterday.

Yes, that is very likely. Adding a line similar to https://github.com/bitcoindevkit/bdk-reserves/blob/main/.github/workflows/cont_integration.yml#L47 will likely fix it. 

I just checked. BDK 0.29 stuck with MSRV 1.57  which becomes funny to maintain. BDK 1 has MSRV 1.63 which should be easier.

**Review by ulrichard — APPROVED** [review]
LGTM
Thanks a lot for improving this.

File: src/reserves.rs [hunk 1]
```diff
@@ -192,7 +187,7 @@ pub fn verify_proof(
     psbt: &PSBT,
     message: &str,
     outpoints: Vec<(OutPoint, TxOut)>,
-    network: Network,
+    _network: Network,
```
[the remark this answers is not in the document]
Comment (Ademan) [reply src/reserves.rs:190]: I wasn't sure if/when you wanted to bump to the next BDK version, I had that in my next patch set which makes a number of more significant changes. I can include it here instead though.

File: src/reserves.rs [hunk 2]
```diff
@@ -258,32 +253,14 @@ pub fn verify_proof(
...
-    }
-    .script_pubkey();
+    let out_script_unspendable = Script::new_p2pkh(&pkh);
+
     if tx.output[0].script_pubkey != out_script_unspendable {
         return Err(ProofError::InvalidOutput);
     }
 
     let serialized_tx = serialize(&tx);
-    // Verify the challenge input
```
[the remark this answers is not in the document]
Comment (Ademan) [reply src/reserves.rs:271]: Yeah I was hesitant to remove a test, but I'm pretty certain this is redundant. In bip-0127 verification is done over transactions, where the witness_utxo wouldn't be available anyway.

File: src/reserves.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ademan) [reply src/reserves.rs:190]: Yeah my intention was to only make backwards compatible changes in this patch set

**Pull request merged**